### ISSUE TITLE
No overload variant of "__getitem__" of "DataFrame" matches argument type "Hashable"  [call-overload]

### ISSUE THREAD

**Issue by randolf-scholz** [pandas-dev/pandas-stubs#659]
**Describe the bug**

Possibly regression from #70.

```python
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})

for col, dtype in df.dtypes.items():
    df[col]  # No overload variant of "__getitem__" of "DataFrame" matches argument type "Hashable"  [call-overload]
```

I would recommend adding a unit test that checks whether `DataFrame.__getitem__` (and `DataFrame.loc.__getitem__`) are type-compatible with `DataFrame.columns` and `DataFrame.dtypes`.

 - Ubuntu 22.04
 - python 3.10.6
 - mypy  1.2.0
 - pandas                              2.0.1
 - pandas-stubs                        1.5.3.230304

**Comment by Dr-Irv** [comment]
I think this is fixed in pandas-stubs 2.0.0.230412 .  Can you test with that?

**Comment by randolf-scholz** [comment]
yep its fixed!
